halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/v2/KubernetesV2FiatService.java

```diff
@@ -49,32 +45,23 @@ public boolean isEnabled(DeploymentConfiguration deploymentConfiguration) {
     return deploymentConfiguration.getSecurity().getAuthz().isEnabled();
   }
 
-  @Override
-  public List<Profile> getProfiles(DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints) {
-    List<Profile> profiles = super.getProfiles(deploymentConfiguration, endpoints);
-
-    if (hasServiceOverrides(deploymentConfiguration)) {
-      SpinnakerRuntimeSettings serviceOverrides = new SpinnakerRuntimeSettings();
-      serviceOverrides.setServiceSettings(Type.CLOUDDRIVER, endpoints.getServiceSettings(Type.CLOUDDRIVER_RO).withOnlyBaseUrl());
-      String filename = "fiat-overrides.yml";
-      String path = Paths.get(getConfigOutputPath(), filename).toString();
-      Profile profile = getSpinnakerProfileFactory().getProfile(filename, path, deploymentConfiguration, serviceOverrides);
-      profiles.add(profile);
-    }
-
-    return profiles;
-  }
-
   @Override
   public ServiceSettings defaultServiceSettings(DeploymentConfiguration deploymentConfiguration) {
-    if (hasServiceOverrides(deploymentConfiguration)) {
-      return new Settings(Arrays.asList("overrides", "local"));
-    }
-    return new Settings();
+    return new Settings(getActiveSpringProfiles(deploymentConfiguration));
   }
 
-  private boolean hasServiceOverrides(DeploymentConfiguration deployment) {
+  @Override
+  protected boolean hasServiceOverrides(DeploymentConfiguration deployment) {
     HaServices haServices = deployment.getDeploymentEnvironment().getHaServices();
     return haServices.getClouddriver().isEnabled();
   }
+
+  @Override
+  protected SpinnakerRuntimeSettings getServiceOverrides(DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints) {
+    SpinnakerRuntimeSettings serviceOverrides = new SpinnakerRuntimeSettings();
+    if (endpoints.serviceIsEnabled(Type.CLOUDDRIVER_RO)) {
```

Comment:
Depending on how many other services we support alternate types for, it may be worth a refactor in the future to make this more generic. Maybe add a TODO?